### PR TITLE
feat: Add floated course page and faculty controller

### PR DESCRIPTION
**Pull request by divs4debu**

### CODE REVIEW

**Comment by divs4debu** [comment]
@iamareebjamal Please Review

File: src/main/java/in/ac/amu/zhcet/data/service/FacultyService.java [hunk 1]
```diff
@@ -59,4 +66,8 @@ public void updateDepartment(String id, Department department) {
         facultyRepository.save(facultyMember);
     }
 
+    public List<FloatedCourse> getByFaculty(FacultyMember facultyMember) {
```
Comment (iamareebjamal) [inline src/main/java/in/ac/amu/zhcet/data/service/FacultyService.java]: This should be in floated course service

File: src/main/resources/templates/department.html [hunk 2]
```diff
@@ -39,11 +39,11 @@ <h4 class="card-title" id="from-actions-top-left">Add Course</h4>
                                 <h4 class="form-section"><i class="icon-head"></i> Course Info</h4>
                                 <br />
                                 <div class="mui-textfield mui-textfield--float-label">
-                                    <input id="code" name="code" class="mtrl" type="text" th:field="*{code}" required/>
+                                    <input id="code" name="code" class="mtrl" type="text" th:field="*{code}"/>
                                     <label class="label-control mtrl" for="code">Course Code</label>
                                 </div>
                                 <div class="mui-textfield mui-textfield--float-label">
-                                    <input id="title" name="title" class="mtrl" type="text" th:field="*{title}" required/>
```
Comment (iamareebjamal) [inline src/main/resources/templates/department.html]: Both these fields are required

File: src/main/resources/templates/fragments/navigation.html [hunk 3]
```diff
@@ -58,6 +58,12 @@
                     <span data-i18n="nav.page_layouts.main" class="menu-title">Login</span>
                 </a>
             </li>
+            <li class="nav-item" sec:authorize="hasRole('ROLE_FACULTY')">
+                <a th:href="@{/faculty/courses}" href="floatedCourse.html">
+                    <i class="icon-stack-2"></i>
+                    <span data-i18n="nav.page_layouts.main" class="menu-title">Floated</span>
```
Comment (iamareebjamal) [inline src/main/resources/templates/fragments/navigation.html]: Change to Floated Courses

**Comment by divs4debu** [comment]
Ok

**Comment by divs4debu** [comment]
@iamareebjamal Please Review

**Pull request merged**